lib/matplotlib/colors.py

```diff
@@ -955,6 +955,176 @@ def scaled(self):
+                                    mask=np.ma.getmask(result))
+        if is_scalar:
+            result = np.atleast_1d(result)[0]
+        return result
+
+    def autoscale_None(self, A):
+        """
+        Parameters:
+        -----------
+
+        A : tuple
+            data used for autoscaling, if appropriate.
+
+        If ``norm.vmin` is None, or ``norm.vmax`` are none, use
+        the min and max of ``A`` for the endpoints of the linear mapping.
+        """
+
+        # allow autoscaling with the data if the user specifies
+        # vmin or vmax is None.  Note we never set vmax/vmin in the class
+        # except at init.
+        vmin = self.vmin
+        if vmin is None:
+            vmin = np.ma.min(A)
+        vmax = self.vmax
+        if vmax is None:
+            vmax = np.ma.max(A)
+
+        if vmin > vmax:
+            raise ValueError('vmin must be less than or equal to vmax')
+
+        if self._data_points is None:
+            self._data_points = np.asarray([vmin, vmax])
+        if self._norm_points is None:
+            N = len(self._data_points)
+            self._norm_points = np.linspace(0, 1, N)
+
+        self._data_points[0] = vmin
+        self._data_points[-1] = vmax
+        self.vmin = vmin
+        self.vmax = vmax
+
+
+class DivergingNorm(PiecewiseLinearNorm):
+    def __init__(self, vmin=None, vcenter=None, vmax=None):
+        """
+        Normalize data with a set center.
+
+        Useful when mapping data with an unequal rates of change around a
+        conceptual center, e.g., data that range from -2 to 4, with 0 as
+        the midpoint.
+
+        Parameters
+        ----------
+        vmin : float, optional
+            The data value that defines ``0.0`` in the normalization.
+            Defaults to the min value of the dataset.
+        vcenter : float, optional
+            The data value that defines ``0.5`` in the normalization.
+            If not defined, the normalization just linearly maps between
+            *vmin* and *vmax*.
+        vmax : float, optional
+            The data value that defines ``1.0`` in the normalization.
+            Defaults to the the max value of the dataset.
+
+        Examples
+        --------
+        This maps data value -4000 to 0., 0 to 0.5, and +10000 to 1.0; data
+        between is linearly interpolated::
+
+            >>> import matplotlib.colors as mcolors
+            >>> offset = mcolors.DivergingNorm(vmin=-4000.,
+                                               vcenter=0., vmax=10000)
+            >>> data = [-4000., -2000., 0., 2500., 5000., 7500., 10000.]
+            >>> offset(data)
+            array([0., 0.25, 0.5, 0.625, 0.75, 0.875, 1.0])
+        """
+
+        # vmin/vmax will set the first and last value, so they don't
+        # matter very much.
+        if vcenter is None:
+            stops = [0, 1]
+        else:
+            if not vcenter == 0:
+                stops = [vcenter * 0.75, vcenter, vcenter * 1.5]
```

Comment:
where does this 0.75 and 1.5 come from?

Reply:
Just a placeholder for when these get set later...

Reply:
see comment below (init to None? don't init to "seemingly" valid values which can later be accidentally used)

Reply:
Because PiecewiseLinearNorm checks *all* the values (i.e. there could be 100 of them) for monotonicity, and therefore if we want to call it's init, we have to send a monotonic list of numbers.

Reply:
PiecewiseLinearNorm's init seems to explicitly cater for initializing with None?